### PR TITLE
Set healing thresholds for necro healing

### PR DESCRIPTION
**Pull request by FluxBorn**
update combat trainer to allow user to set a healing threshold to avoid healing scratches if you so choose. Updated base yaml with the setting defaulting to heal everything.

### CODE REVIEW

**Comment by izawwlgood** [comment]
Can confirm this works as intended, and will likely be useful for Empath combat healing as well.

File: combat-trainer.lic [hunk 1]
```diff
@@ -428,10 +428,12 @@ class LootProcess
       if @necro_heal && !game_state.necro_casting?
         game_state.wounds = check_health
         echo "Severity to Wounds: #{game_state.wounds}" if $debug_mode_ct
-        unless game_state.wounds.empty?
-          do_necro_ritual(mob_noun, 'consume', game_state)
-          return false
-        end
+		if @heal_threshold <= game_state.wounds.keys.max
```
Comment (jandersson) [inline combat-trainer.lic]: any chance you can set your indents to 2 spaces as per the Ruby style?

Comment (jandersson) [reply combat-trainer.lic]: Rubocop will fix this automagically for you if you're interested: https://github.com/bbatsov/rubocop

Running `rubocop -a` in the scripts directory will autocorrect all script files

Comment (FluxBorn) [reply combat-trainer.lic]: Nifty, I will use this next time.

File: combat-trainer.lic [hunk 2]
```diff
@@ -428,9 +428,11 @@ class LootProcess
       if @necro_heal && !game_state.necro_casting?
         game_state.wounds = check_health
         echo "Severity to Wounds: #{game_state.wounds}" if $debug_mode_ct
-        unless game_state.wounds.empty?
-          do_necro_ritual(mob_noun, 'consume', game_state)
-          return false
+        if @heal_threshold <= game_state.wounds.keys.max
```
Comment (rpherbig) [inline combat-trainer.lic]: Does this work if you have no wounds?

My understanding is that `game_state.wounds.keys.max` would be `nil` in that case. Comparing `1 <= nil` should result in an exception.

File: combat-trainer.lic [hunk 3]
```diff
@@ -428,9 +428,11 @@ class LootProcess
...
+          unless game_state.wounds.empty?
```
Comment (rpherbig) [inline combat-trainer.lic]: Is this conditional still necessary?

File: profiles/base.yaml [hunk 4]
```diff
@@ -82,6 +82,7 @@ buff_nonspells:
 dance_actions_stealth:
 thanatology:
 necromancer_healing:
+heal_threashold: 1
```
Comment (rpherbig) [inline profiles/base.yaml]: Should this be `heal_threshold`?

File: profiles/base.yaml [hunk 5]
```diff
@@ -82,6 +82,7 @@ buff_nonspells:
 dance_actions_stealth:
 thanatology:
 necromancer_healing:
```
Comment (rpherbig) [inline profiles/base.yaml]: The setting name `heal_threshold` is rather generic. What would you think about putting `necro` or `necromancer` somewhere in the name?

Or maybe it should be a sub-setting of `thanatology` or `necromancer_healing`? @jandersson might have a better idea on that one...

File: profiles/base.yaml (hunk 4 above)
Comment (FluxBorn) [reply profiles/base.yaml]: Correct, updating the typo

File: combat-trainer.lic (hunk 2 above)
Comment (FluxBorn) [reply combat-trainer.lic]: very true, and I missed it in testing. I will switch the logic around to check for this.

File: combat-trainer.lic (hunk 3 above)
Comment (FluxBorn) [reply combat-trainer.lic]: I switched the logic around to have the unless check for nill before moving to the if > threshold

**Comment by izawwlgood** [comment]
If you can use heal_threshold for Empath battle healing too, it might make sense to simply use the same flag? If there's potential crosstalk problems, then yeah, necro_self_heal_threshold or somesuch?

File: profiles/base.yaml (hunk 5 above)
Comment (FluxBorn) [reply profiles/base.yaml]: So I may add this to the regular empath healing logic as well, so not sure at that point if it should be a separate variable, or if all healing yaml should be rolled up..

Comment (rpherbig) [reply profiles/base.yaml]: Maybe `self_heal_threshold`?

I'm concerned it could be confused with the existing `health_threshold` setting

Comment (FluxBorn) [reply profiles/base.yaml]: Hmm wound_level_threshold?

Comment (FluxBorn) [reply profiles/base.yaml]: I suppose making it a sub of necromancer_healing would be better.

Comment (FluxBorn) [reply profiles/base.yaml]: Moved under necromancer_healing, and renamed wound_level_threshold

File: combat-trainer.lic [hunk 6]
```diff
@@ -428,9 +432,12 @@ class LootProcess
       if @necro_heal && !game_state.necro_casting?
         game_state.wounds = check_health
         echo "Severity to Wounds: #{game_state.wounds}" if $debug_mode_ct
+        echo "wound_level_threshold: #{@wound_level_threshold}"
```
Comment (rpherbig) [inline combat-trainer.lic]: Should this be conditional on `if $debug_mode_ct`?

File: combat-trainer.lic [hunk 7]
```diff
@@ -428,9 +432,12 @@ class LootProcess
...
         unless game_state.wounds.empty?
-          do_necro_ritual(mob_noun, 'consume', game_state)
-          return false
+		      if @wound_level_threshold <= game_state.wounds.keys.max
```
Comment (rpherbig) [inline combat-trainer.lic]: It looks like there is unusual indentation here, could you run Rubocop again?

**Comment by rpherbig** [comment]
Is this PR adding a blank file (`necroheal.lic`)?

File: profiles/base.yaml (hunk 5 above)
Comment (FluxBorn) [reply profiles/base.yaml]: let me know if the new variable name and location work for you.

**Pull request merged**